R/module_data_summary.R

```diff
@@ -95,60 +80,74 @@ srv_data_summary <- function(id, teal_data) {
         } else if (is.null(summary_table_out)) {
           "no datasets to show"
         } else {
+          is_unsupported <- apply(summary_table(), 1, function(x) all(is.na(x[-1])))
+          summary_table_out[is.na(summary_table_out)] <- ""
           body_html <- apply(
             summary_table_out,
             1,
             function(x) {
-              tags$tr(
-                tagList(
-                  tags$td(
-                    if (all(x[-1] == "")) {
-                      icon(
-                        name = "fas fa-exclamation-triangle",
-                        title = "Unsupported dataset",
-                        `data-container` = "body",
-                        `data-toggle` = "popover",
-                        `data-content` = "object not supported by the data_summary module"
-                      )
-                    },
-                    x[1]
-                  ),
-                  lapply(x[-1], tags$td)
+              is_supported <- !all(x[-1] == "")
+              if (is_supported) {
+                tags$tr(
+                  tagList(
+                    tags$td(x[1]),
+                    lapply(x[-1], tags$td)
+                  )
                 )
-              )
+              }
             }
           )
 
-          header_labels <- names(summary_table())
+          header_labels <- tools::toTitleCase(names(summary_table_out))
+          header_labels[header_labels == "Dataname"] <- "Data Name"
           header_html <- tags$tr(tagList(lapply(header_labels, tags$td)))
 
           table_html <- tags$table(
             class = "table custom-table",
             tags$thead(header_html),
             tags$tbody(body_html)
           )
-          table_html
+          div(
+            table_html,
+            if (any(is_unsupported)) {
+              p(
+                class = c("pull-right", "float-right", "text-secondary"),
+                icon(
+                  name = "fas fa-circle-info",
+                  title = paste(
+                    sep = "",
+                    collapse = ", ",
+                    shQuote(summary_table()[is_unsupported, "dataname"]),
+                    " (",
+                    vapply(summary_table()[is_unsupported, "dataname"], function(x) typeof(teal_data()[[x]]), character(1L)),
+                    ")"
+                  )
+                ),
+                sprintf("+%s unfilterable dataset(s)", sum(is_unsupported))
```

Comment:
```suggestion
                sprintf("+%s unfilterable object(s)", sum(is_unsupported))
```

Since we include non-dataset objects here I liked the previous naming where we called it just object.